fili-core/src/main/java/com/yahoo/bard/webservice/data/config/table/BaseTableLoader.java

```diff
@@ -57,33 +60,23 @@ protected BaseTableLoader() {
         this(DateTimeZone.UTC);
     }
 
-    @Override
-    public abstract void loadTableDictionary(ResourceDictionaries dictionaries);
-
     /**
-     * Builds a table group.
-     * <p>
-     * Builds and loads the physical tables for the physical table definitions as well.
-     *
-     * @param logicalTableName  The logical table for the table group
-     * @param apiMetrics  The set of metric names surfaced to the api
-     * @param druidMetrics  Names of druid datasource metric columns
-     * @param tableDefinitions  A list of config objects for physical tables
-     * @param dictionaries  The container for all the data dictionaries
+     * Load user configured tables into resource dictionary.
      *
-     * @return A table group binding all the tables for a logical table view together.
+     * @param dictionaries dictionary to be loaded with user configured tables
      *
-     * @deprecated logicalTableName is not used in TableGroup
+     * @deprecated should use the same method that takes an additional DataSourceMetadataService argument instead
      */
     @Deprecated
-    public TableGroup buildTableGroup(
-            String logicalTableName,
-            Set<ApiMetricName> apiMetrics,
-            Set<FieldName> druidMetrics,
-            Set<PhysicalTableDefinition> tableDefinitions,
-            ResourceDictionaries dictionaries
+    public abstract void loadTableDictionary(ResourceDictionaries dictionaries);
+
+    @Override
+    public void loadTableDictionary(
+            ResourceDictionaries dictionaries,
+            DataSourceMetadataService metadataService
     ) {
-        return buildTableGroup(apiMetrics, druidMetrics, tableDefinitions, dictionaries);
+        this.metadataService = metadataService;
+        this.loadTableDictionary(dictionaries);
```

Comment:
I think simply removing the deprecated tag to the `loadTableDictionary` method that people are using today should fix all the problem, since user should not care and does not have to touch the `metadataService`, instead fili takes care of it in its `AbstractBinderFactory`.